tests/torch/test_hybrid_converter.py

```diff
@@ -240,3 +272,85 @@ def test_invalid_model():
     # Attempt to create a HybridFHEModel with an invalid model type and expect a TypeError
     with pytest.raises(TypeError, match="The model must be a PyTorch or Brevitas model."):
         HybridFHEModel(invalid_model, module_names="sub_module")
+
+
+@pytest.mark.parametrize("n_hidden", [512, 2048])
+def test_hybrid_glwe_correctness(n_hidden):
+    """Tests that the GLWE backend produces correct results for the hybrid model."""
+
+    num_samples = 200
+
+    def prepare_data(x, y, test_size=0.1, random_state=42):
+        x_train, x_test, y_train, y_test = train_test_split(
+            x, y, test_size=test_size, random_state=random_state
+        )
+        x_train = torch.tensor(x_train, dtype=torch.float32)
+        x_test = torch.tensor(x_test, dtype=torch.float32)
+        y_train = torch.tensor(y_train, dtype=torch.long)
+        y_test = torch.tensor(y_test, dtype=torch.long)
+        return x_train, x_test, y_train, y_test
+
+    # Generate synthetic 2D data
+    x1_data, y1_data = make_moons(n_samples=num_samples, noise=0.2, random_state=42)
+
+    # Prepare data
+    x1_train, x1_test, y1_train, y1_test = prepare_data(x1_data, y1_data)
+
+    model = FCSmall(2, torch.nn.ReLU, hidden=n_hidden)
+    optimizer = torch.optim.Adam(model.parameters())
+
+    num_epochs = 100
+    model.train()
+    for _ in range(num_epochs):
+        optimizer.zero_grad()
+        outputs = model(x1_train)
+        loss = torch.nn.functional.cross_entropy(outputs, y1_train)
+        loss.backward()
+        optimizer.step()
+
+    model.eval()
+
+    param_names = []
+    for k, p in model.named_modules():
+        if isinstance(p, torch.nn.Linear):
+            param_names.append(k)
+
+    y_torch = model(x1_test).detach().numpy()
+    hybrid_local = HybridFHEModel(model, param_names)
+
+    # This internal flag tells us whether all the layers
+    # were linear and were replaced with the GLWE backend
+    assert hybrid_local._all_layers_are_pure_linear  # pylint: disable=protected-access
+
+    hybrid_local.compile_model(x1_train, n_bits=10)
+
+    y_qm = hybrid_local(x1_test, fhe="disable").numpy()
+    y_hybrid_torch = hybrid_local(x1_test, fhe="torch").detach().numpy()
+    y_glwe = hybrid_local(x1_test, fhe="execute").numpy()
+
+    y1_test = y1_test.numpy()
+    acc_fp32 = numpy.sum(numpy.argmax(y_torch, axis=1) == y1_test)
+    acc_qm = numpy.sum(numpy.argmax(y_qm, axis=1) == y1_test)
+    acc_glwe = numpy.sum(numpy.argmax(y_glwe, axis=1) == y1_test)
+
+    # These two should be exactly the same
+    assert numpy.all(numpy.allclose(y_torch, y_hybrid_torch, rtol=1, atol=0.001))
+
+    # The clear quantization vs fp32 test has more tolerance
+    threshold_fhe = 0.01
```

Comment:
experimentally adjusted. 

the accuracy is fine even with this threshold on the logits